src/Datadog.Trace/Util/SpanIdGenerator.cs

```diff
@@ -67,5 +69,15 @@ public ulong CreateNew()
 
             return (ulong)value & 0x7FFFFFFFFFFFFFFF;
         }
+
+        public BigInteger CreateNew128Bit()
+        {
+            var array = new byte[16];
+            using var generator = RandomNumberGenerator.Create();
```

Comment:
Using the crypto one may be too expensive, look at https://github.com/dotnet/runtime/pull/46890 and the RandomNumberGenerator that they added there.

Reply:
This is a leftover from previous POC, I will get rid of it in the next commit.